identity/app/actions/AuthenticatedActions.scala

```diff
@@ -42,8 +40,30 @@ class AuthenticatedActions(
   def sendUserToReauthenticate(request: RequestHeader): Result =
     redirectWithReturn(request, "/reauthenticate")
 
-  def authAction: AuthenticatedBuilder[AuthenticatedUser] =
-    new AuthenticatedBuilder(authService.authenticatedUserFor, anyContentParser, sendUserToSignin)
+  def sendUserToRegister(request: RequestHeader) : Result =
+    redirectWithReturn(request, "/register")
+
+  def authRefiner: ActionRefiner[Request, AuthRequest] = new ActionRefiner[Request, AuthRequest] {
+    override val executionContext = ec
+
+    def refine[A](request: Request[A]) =
+      authService.authenticatedUserFor(request) match {
+        case Some(authenticatedUser) => Future.successful(Right(new AuthenticatedRequest(authenticatedUser, request)))
+        case None =>
+          // If an email query param exists we want to check the DB for the user and redirect them to signin/register
+          request.getQueryString("email") match {
+            case None => Future.successful(Left(sendUserToSignin(request)))
+            case Some(email) =>
+              identityApiClient.userFromQueryParam(email, "emailAddress").map {
+                case Right(userExists) => Left(sendUserToSignin(request))
```

Comment:
I assume when the user doesn't exist this case won't be triggered?  It's not that clear whether `userExists` will be some kind of default empty `User` or it ends up as a `Left`.